lib/workers/global/config/parse/file.ts

```diff
@@ -1,8 +1,28 @@
+import { readFile } from 'fs-extra';
+import { load } from 'js-yaml';
+import JSON5 from 'json5';
 import upath from 'upath';
 import { migrateConfig } from '../../../../config/migration';
-import type { AllConfig } from '../../../../config/types';
+import type { AllConfig, RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 
+export async function getParsedContent(file: string): Promise<RenovateConfig> {
+  switch (upath.extname(file)) {
+    case '.yaml':
+    case '.yml':
+      return load(await readFile(file, 'utf8'), {
+        json: true,
+      }) as RenovateConfig;
+    case '.json5':
+      return JSON5.parse(await readFile(file, 'utf8'));
+    default: {
+      // .json and .js
```

Comment:
```suggestion
      // .js
```

I think we should add a case for `js` files and throw an error on default.

Reply:
Do you think this might break things for people who have been defining their `RENOVATE_CONFIG_FILE` without a .js extension? For example, as with the default:
https://github.com/renovatebot/renovate/blob/6f9f9689333f3757cf5d64c3b4567cf6fc160477/lib/workers/global/config/parse/file.ts#L27

So maybe I could change 
```diff
-  let configFile = env.RENOVATE_CONFIG_FILE || 'config';
+  let configFile = env.RENOVATE_CONFIG_FILE || 'config.js';
```

And add a breaking change trailer?

Reply:
Oh, not thought about that. Maybe we should try to resolve the extensionless file to `js`  or `json` , we need to do this for esm later anyways, because extensions are required for esm.

Reply:
Now I'm not sure if renovate should also explicitly exit non-zero on invalid file types or ignore it? I tried to do the first option.

Reply:
From what I see in https://github.com/renovatebot/renovate/issues/7620, it should probably fail also on invalid filetypes, so I've added this and some tests for that scenario.